src/Integration.Vsix/ErrorList/IssuesSnapshot.cs

```diff
@@ -60,17 +60,19 @@ internal interface IIssuesSnapshot : ITableEntriesSnapshot
         /// <returns></returns>
         IEnumerable<IAnalysisIssueLocationVisualization> GetLocationsVizsForFile(string filePath);
 
-        /// <summary>
-        /// Notifies the snapshot that some part of the contained data has changed and that it should
-        /// increment its version so the Error List recognises it as having changed
-        /// </summary>
-        void IncrementVersion();
-
         /// <summary>
         /// Create and return an updated version of an existing snapshot, where the
         /// issues are the same but the source file has been renamed
         /// </summary>
         IIssuesSnapshot CreateUpdatedSnapshot(string analyzedFilePath);
+
+        /// <summary>
+        /// Notifies the snapshot that some part of the contained data has changed.
+        /// The snapshot will then decide whether:
+        /// 1. Return the same snapshot instance with an incremented version so the Error List recognises it as having changed
+        /// 2. Return a new snapshot instance, excluding non-navigable issues
+        /// </summary>
+        IIssuesSnapshot GetUpdatedSnapshot();
```

Comment:
A couple of points:
* I think the detail about exactly what changes is too implementation-specific for the interface. It's important, but should be in the class instead.
* we should capture the point we discussed about object identities

I was going to try to add a GitHub `suggestion`, but it doesn't look like they can span multiple lines so I've just typed it below:

```
/// <Summary>
/// Notifies the snapshot that some part of the contained data has changed.
/// An updated snapshot will be returned that the Error List will recognise as having changed.
/// Any non-navigable items will have been removed.
/// </summary>
/// <remarks>
/// Note about object identities:
/// * the caller *cannot* rely on the identity of the snapshot object being preserved
/// * the caller *can* rely on the identity of individual issues being preserved
/// </remarks>
```